keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/PartialsOf.java

```diff
@@ -0,0 +1,38 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.diff;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexId;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@AllArgsConstructor
+public class PartialsOf {
+
+    @Getter
+    private final Vertex<?> of;
+
+    private final Map<VertexId, PartialDerivative> partials;
+
+    public PartialDerivative withRespectTo(Vertex vertex) {
+        return withRespectTo(vertex.getId());
+    }
+
+    public PartialDerivative withRespectTo(VertexId id) {
+        return partials.get(id);
+    }
+
+    public Map<VertexId, DoubleTensor> asMap() {
+        final Map<VertexId, DoubleTensor> tensorMap = new HashMap<>();
+
+        for (Map.Entry<VertexId, PartialDerivative> entry : partials.entrySet()) {
+            tensorMap.put(entry.getKey(), entry.getValue().get());
```

Comment:
Given this is in the hot path of reverse mode auto-diff - I'm not sure I love having to do this translation from one map type to another - just seems to be adding busy work to avoid other people seeing PartialDerivatives rather than DoubleTensors ?

Reply:
I tried avoiding this earlier but gave up. I've refactored it to to the conversion at the point the double tensor is used. This avoids the double iteration and the new object creation.